go/vt/vtgate/vindexes/region_json.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vindexes
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"io/ioutil"
+
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/key"
+	"vitess.io/vitess/go/vt/log"
+)
+
+var (
+	_ MultiColumn = (*RegionJson)(nil)
+)
+
+func init() {
+	Register("region_json", NewRegionJson)
+}
+
+// RegionMap is used to store mapping of country to region
+type RegionMap map[string]uint64
+
+// RegionJson defines a vindex that uses a lookup table.
+// The table is expected to define the id column as unique. It's
+// Unique and a Lookup.
+type RegionJson struct {
```

Comment:
Why not wrap this around RegionExperimental? Then you don't have to duplicate the `Verify` function (which to me looks identical to the implementation in RegionExperimental).

Reply:
@deepthi To be honest this is the first Go I've written, so I'm not exactly sure what the syntax do do this would be (or even what the official term for "wrapping around" is, so I'm not finding anything in the documentation).

Might you have a pointer to an example of this sort of wrapping, or to Go documentation that might help me make this change?

Reply:
You can see it in the version you adapted this from: 
```
type RegionVindex struct {
	regionMap RegionMap
	*RegionExperimental
}
```
There you will also see how this is being initialized.

Reply:
@deepthi ah ok, I removed it as that's what the other files seemed to be doing.

Reply:
@deepthi I tried copy/pasting the code back in.  It seems to be failing due to not finding 'ConsistentLookupUnique' in 'RegionExperimental', which seems to have been removed.  Should I remove the `if` check?  Not sure of the entire ecosystem you have/other changes in the works.

Reply:
you may as well leave it alone then. we can fix it up later if we want.

Reply:
The other thing is that it will be nice to have a test for this. Or at least make sure it is working for you before we merge.

Reply:
@deepthi I've undone the update with `RegionExperimental`.  Currently testing with our infrastructure to see if it's working for me.

Reply:
Just saw this. In case you were wondering why it didn't work: you can't reuse the `Verify` function from RegionExperimental because it invokes `Map`, and that will go to RegionExperimental's `Map`.